aq/engines.py

```diff
@@ -23,17 +23,22 @@ def __init__(self, options=None):
         self.options = options if options else {}
         self.debug = options.get('--debug', False)
 
+        self.profile = options.get('--profile', None)
+        self.region = options.get('--region', None)
         self.table_cache_ttl = int(options.get('--table-cache-ttl', 300))
         self.last_refresh_time = defaultdict(int)
 
-        self.boto3_session = boto3.Session()
+        self.boto3_session = boto3.Session(profile_name=self.profile)
         # dash (-) is not allowed in database name so we use underscore (_) instead in region name
         # throughout this module region name will *always* use underscore
-        if self.boto3_session.region_name:
+        if self.region:
+            self.default_region = self.region.replace('-', '_')
+        elif self.boto3_session.region_name and self.region is None:
```

Comment:
@kubihie the check for `self.region is None` seems redundant to me here, isn't it?